### PR TITLE
Some notes about developing on Windows

### PR DESCRIPTION
**Pull request by ezyang**
CC @yf225 @cpuhrsch @peterjc123

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -235,6 +235,61 @@ If you are working on the CUDA code, here are some useful CUDA debugging tips:
...
+Occasionally, you will write a patch which works on Linux, but fails CI on Windows.
+There are a few aspects in which MSVC (the Windows compiler toolchain we use) is stricter
+than Linux, which are worth keeping in mind when fixing these problems.
+
+1. Symbols are NOT exported by default on Windows; instead, you have to explicitly
+   mark a symbol as exported/imported in a header file with `__declspec(dllexport)` /
+   `__declspec(dllimport)`.  We have codified this pattern into a set of macros
+   which follow the convention `*_API`, e.g., `AT_API` inside ATen. (Every separate
+   shared library needs a unique macro name, because symbol visibility is on a per
+   shraed library basis.)
```
Comment (yf225) [inline CONTRIBUTING.md]: nit: "shared"

**Comment by yf225** [comment]
@ezyang I am also working on a guide on how to SSH into the Windows EC2 machines and do development work (using vim / emacs, building PyTorch using the batch file, using GDB). I will add it to `CONTRIBUTING.md` when ready.

**Comment by peterjc123** [comment]
@ezyang @yf225 Great!

**Review by orionr — APPROVED** [review]
Excellent!

**Review by yf225 — APPROVED** [review]

**Pull request merged**